polkadot/xcm/xcm-executor/src/traits/asset_exchange.rs

```diff
@@ -37,6 +37,15 @@ pub trait AssetExchange {
 		want: &Assets,
 		maximal: bool,
 	) -> Result<AssetsInHolding, AssetsInHolding>;
+
+	/// Handler for quoting the exchange price of two assets.
+	///
+	/// - `asset1` The first asset.
+	/// - `asset2` The second asset.
+	/// - `maximal`: If `true`, then all of `asset1` should be used.
+	fn quote_exchange_price(_asset1: &Asset, _asset2: &Asset, _maximal: bool) -> Option<u128> {
+		None
```

Comment:
Do we use the new func in any existing functionality? So if client has it's implementation the things that used to work, will silently stop working. 
Also semantically this default impl does not look correct to me.

Reply:
The new function is only used in the new functionality added in https://github.com/paritytech/polkadot-sdk/pull/5131, if you don't implement it then nothing breaks, it keeps on working as it used to.

Semantically, this function returning `None` means the assets are not exchangeable.
While this may not be true, the reason we have the default impl is to not make this a breaking change.
Most users are not going to use this, the ones that do will override this anyways.
What we do not want is to break the code of users that are not going to use this.

Reply:
It's true that if some team has their own implementation of the `AssetExchanger`, like I know Hydration does, then this would affect them, but only in that they wouldn't receive the benefits of https://github.com/paritytech/polkadot-sdk/pull/5131, nothing would break.

Reply:
So for those users the new feature will be silently disabled. Right, it means not exchangeable, but in reality it might be not true (for any real impl it is not true). I would rather be correct here. 
If you decide to keep this default impl, please leave the comment with the reason why. Otherwise it would be hard to reason it. It does not serve here as a general default impl, but only for a backward compatibility.

Reply:
It might be better to remove the default implementation and just have this be a breaking change then. The only reason why we would add this is so that we don't have a breaking change. But since this will come out in a new major version of a crate it should be a conscious decision by the teams upgrading. Also not every team implements this, so the impact shouldn't be that big.